pallets/rewards/src/lib.rs

```diff
@@ -268,14 +267,18 @@ pub mod pallet {
 
 				Groups::<T>::try_mutate(group_id, |group| {
 					StakeAccounts::<T>::try_mutate(account_id, currency_id, |account| {
-						T::Currency::hold(currency_id.1, account_id, amount)?;
+						if !T::Currency::can_hold(currency_id.1, account_id, amount) {
```

Comment:
`thought`

I'm not a fan of creating type aliases for tuples and then reference the fields like we do here with `currency_id.1`. I'd rather have the current type alias become a struct called `DomainCurrency` that has a `DomainId` and a `CurrencyId`. Then here we could either deconstruct that struct and just reference it as `currency_id` or directly as `domain_currency.currency_id`.